### PR TITLE
Adds metering to WASM bytecode as part of conversion process per @axic

### PR DESCRIPTION
**Pull request by lrettig**

### CODE REVIEW

**Comment by lrettig** [comment]
Note that this PR includes #395, #396, needed to merge those in before updating `convert_wast.py`

**Comment by lrettig** [comment]
Cf https://github.com/ewasm/wasm-metering/pull/7

**Comment by winsvega** [comment]
wait why we need a workaround like this. could testeth just compile wasm ? 
https://github.com/ethereum/cpp-ethereum/pull/4754

**Comment by axic** [comment]
Note: @jwasinger's script to compile wast-to-wasm will not be needed once https://github.com/ethereum/cpp-ethereum/pull/4754 is merged.

Also, this PR will not be needed once metering is implemented properly.

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "src/GeneralStateTestsFiller/stEWASMTests/utils/wabt"]
 	path = src/GeneralStateTestsFiller/stEWASMTests/utils/wabt
 	url = https://github.com/webassembly/wabt
+[submodule "src/GeneralStateTestsFiller/stEWASMTests/utils/wasm-metering"]
```
Comment (winsvega) [inline .gitmodules:4]: do we really need submodule?

File: src/GeneralStateTestsFiller/stEWASMTests/utils/README.md [hunk 2]
```diff
@@ -17,7 +17,12 @@ mkdir build
 cd build
 cmake -DBUILD_TESTS=OFF ..
 make -j4
-mv wat2wasm ../..
+```
+
+Install wasm-metering
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stEWASMTests/utils/README.md:22]: whats wasm-metering?

**Comment by axic** [comment]
@winsvega this is all testing code merging into @jwasinger's branch. It is not meant to be merged into master. Please disregard it.

**Comment by jwasinger** [comment]
@lrettig , please squash these commits and I will merge.  also can you remove unwanted changes from this PR?  A lot of files are touched/changed which are seemingly irrelevant to stated the purpose of this PR

**Comment by lrettig** [comment]
Working on it

**Comment by lrettig** [comment]
Squashed, back to you

**Comment by axic** [comment]
@jwasinger can you merge this and push the commits to ewasm/tests ?

**Comment by jwasinger** [comment]
merged here: https://github.com/ewasm/tests/commit/edfcdc903ecb16e02b81b94a93a6b69a71e3c03c

**Pull request closed without merging**